x-pack/test/api_integration/apis/endpoint/whitelist.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import expect from '@kbn/expect/expect.js';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ getService }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertest');
+
+  const sizeOfFixture: number = 1;
+
+  describe('test whitelist api', () => {
+    describe('Tests for whitelist API', () => {
+      beforeEach(() => esArchiver.load('endpoint/whitelist/api_feature'));
+      afterEach(() => esArchiver.unload('endpoint/whitelist/api_feature'));
+
+      it('should return whitelist rules when the GET method is called', async () => {
+        const resp = await supertest.get('/api/endpoint/whitelist').set('kbn-xsrf', 'xxx');
+
+        expect(resp.statusCode).to.equal(200);
+        const responseBody = JSON.parse(resp.text);
+        expect(responseBody.entries.length).to.equal(sizeOfFixture);
+      });
+
+      it('should return the whitelist manifest when the GET method is called', async () => {
+        const resp = await supertest.get('/api/endpoint/manifest').set('kbn-xsrf', 'xxx');
+
+        expect(resp.statusCode).to.equal(200);
+        const responseBody = JSON.parse(resp.text);
+        const cachedWhitelist = responseBody.artifacts['global-whitelist'];
+        // expect(cachedWhitelist.size).to.equal(sizeOfFixture);  // TODO fix me
+        expect(cachedWhitelist.encoding).to.equal('xz');
+        expect(cachedWhitelist.url).to.contain(cachedWhitelist.sha256);
+      });
+
+      it('should insert a whilelist rule into elasticsearch properly', async () => {
```

Comment:
I think it would be better to reorganize this test a bit. If you used a `describe` block instead of nesting the calls, we would be able to more easily tell which part of the rule creation failed if the test were to fail. I suggest doing something like this:
```javascript
describe('upon insertion of a new whitelist rule into elasticsearch', async () => {
  beforeAll(() => console.log('make the POST'));
  it('should add a new rule to the whitelist', async () => {});
  it('whitelist manifest should to be updated correctly', async () => {});
  it('whitelist artifact should have the right contents', async () => {});
});
```

Reply:
That won't work here. The test relies on a sequence of API calls, not just one POST request. I can break this up into a smaller test if you want, but I think it's valuable as is. It tests the entire workflow of creating a whitelist rule, checking the manifest, and downloading from the manifest.

Reply:
Oh I see what you're saying... Yeah we could do that.